cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SwitchesChain.java

```diff
@@ -31,232 +29,173 @@
-        }
+        });
 
-        private static void addAdjacency(Map<String, List<String>> adjacency, int nodeEnd1, int nodeEnd2) {
-            String end1 = String.valueOf(nodeEnd1);
-            String end2 = String.valueOf(nodeEnd2);
-            adjacency.computeIfAbsent(end1, k -> new ArrayList<>()).add(end2);
-            adjacency.computeIfAbsent(end2, k -> new ArrayList<>()).add(end1);
+        if (terminals.size() == 1) {
+            return terminals.get(0);
         }
+        return null;
+    }
+
+    private static Terminal uniqueTerminalAssociatedConductingEquipmentBusBranch(VoltageLevel vl, List<String> configuredBuses) {
+        List<Terminal> terminals = new ArrayList<>();
+
+        vl.getConnectables().forEach(c -> {
+            if (isDiscarded(c)) {
+                return;
+            }
 
-        private static void adjacencyBusBreaker(VoltageLevel vl, String switchId, Map<String, List<String>> adjacency) {
-            vl.getSwitches().forEach(sw -> {
-                if (sw.getId().equals(switchId)) {
+            c.getTerminals().forEach(terminal -> {
+                if (!isSameVoltageLevel((Terminal) terminal, vl)
+                    || !isTerminalInConfiguredBuses((Terminal) terminal, configuredBuses)) {
                     return;
                 }
-                addAdjacency(adjacency, vl.getBusBreakerView().getBus1(sw.getId()), vl.getBusBreakerView().getBus2(sw.getId()));
+                terminals.add((Terminal) terminal);
             });
-        }
+        });
 
-        private static void addAdjacency(Map<String, List<String>> adjacency, Bus busEnd1, Bus busEnd2) {
-            String end1 = busEnd1.getId();
-            String end2 = busEnd2.getId();
-            adjacency.computeIfAbsent(end1, k -> new ArrayList<>()).add(end2);
-            adjacency.computeIfAbsent(end2, k -> new ArrayList<>()).add(end1);
+        if (terminals.size() == 1) {
+            return terminals.get(0);
         }
+        return null;
+    }
 
-        private List<String> getAdjacents(String nodeId) {
-            if (adjacencies.containsKey(nodeId)) {
-                return adjacencies.get(nodeId);
-            }
-            return Collections.<String>emptyList();
+    private static boolean isTerminalInNodes(Terminal terminal, List<Integer> nodes) {
+        return nodes.contains(terminal.getNodeBreakerView().getNode());
+    }
+
+    private static boolean isTerminalInConfiguredBuses(Terminal terminal, List<String> configuredBuses) {
+        return configuredBuses.contains(terminal.getBusBreakerView().getBus().getId());
+    }
+
+    private static boolean isSameVoltageLevel(Terminal terminal, VoltageLevel vl) {
+        return terminal.getVoltageLevel().equals(vl);
+    }
+
+    private static boolean isDiscarded(Connectable<?> connectable) {
+        return connectable.getType() == IdentifiableType.BUSBAR_SECTION;
+    }
+
+    // The best terminal is the terminal associated to the line, branch at the border
```

Comment:
Either put that comment inside the method or use the javadoc syntax
```
/**
 * xxx
 */
```

Reply:
Done